lib/src/components/NxSmallTag/NxSmallTag.scss

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+ @use '../../scss-shared/nx-text-helpers';
+
+.nx-small-tag {
+  @include nx-text-helpers.semi-bold;
+
+  align-items: center;
+  border-radius: 8px;
+  box-sizing: border-box;
+  color: var(--nx-swatch-white);
+  display: inline-flex;
+  font-size: var(--nx-font-size-s);
+  height: 24px;
+  margin: 0 var(--nx-spacing-2x);
+  max-width: 200px;
+  padding: 0 var(--nx-spacing-2x);
+  position: relative;
+  vertical-align: middle;
```

Comment:
Unfortunately, this does not quite produce the desired effect.  See here that it's still too low:
![image](https://github.com/sonatype/sonatype-react-shared-components/assets/3630091/6c4c5ad1-74cb-43fc-832b-94ec6f64c204)

This is the description of what `vertical-align: middle` does:
> Aligns the middle of the element with the baseline plus half the x-height of the parent.

As the x-height means the height of a _lowercase_ x, this is almost never actually useful.  I'm working on a better solution.

Reply:
> As the x-height means the height of a lowercase x, this is almost never actually useful. I'm working on a better solution.
See here: https://github.com/sonatype/sonatype-react-shared-components/compare/RSC-1721_small_tag_component...RSC-1721_small_tag_component_ross

I think an example with `NxSmallTag` inline with other text of varying sizes (and in multiple lines both of other text and other `NxSmallTag`s should be added and tested.

Reply:
Examples added here 219c4dfa4ad15a82dd54c873f0ad99cdb653e6ee, f12fbdfffae05fbf5b4605e0f6bf2981139ef059.

Reply:
> I think an example with NxSmallTag inline with other text of varying sizes (and in multiple lines both of other text and other NxSmallTags should be added and tested.
 
 I've realized that my previous suggested implementation still wasn't quite right - I needed to set the (line) height of `.nx-small-tag` to the _line-height_ of the parent, not the font-size.  See https://github.com/sonatype/sonatype-react-shared-components/compare/RSC-1721_small_tag_component...RSC-1721_small_tag_component_ross again for further suggestions which further improve the alignment.

Reply:
Updated css and snapshots 4e88bd0b9317f7024c506d16793d983546be0df1, 795d274c15bc4a28c7d18c6382020373fe84ea6d.